pulsar-client-cpp/include/pulsar/ConsumerConfiguration.h

```diff
@@ -33,6 +33,7 @@ class PulsarWrapper;
 
 /// Callback definition for non-data operation
 typedef boost::function<void(Result result)> ResultCallback;
+typedef boost::function<void(const Message& msg)> ReceiveCallback;
```

Comment:
however, we should immediately fail the callback if consumer is closing or already closed. so, I have added the result into the callback.